gssapi/raw/ext_rfc5801.pyx

```diff
@@ -0,0 +1,88 @@
+from gssapi.raw.cython_types cimport *
+from gssapi.raw.oids cimport OID
+GSSAPI="BASE"  # This ensures that a full module is generated by Cython
+
+from gssapi.raw.cython_converters cimport c_make_oid
+
+from gssapi.raw.named_tuples import InquireSASLNameResult
+from gssapi.raw.misc import GSSError
+
+cdef extern from "python_gssapi_ext.h":
+    OM_uint32 gss_inquire_saslname_for_mech(
+        OM_uint32 *min_stat,
+        const gss_OID desired_mech,
+        gss_buffer_t sasl_mech_name,
+        gss_buffer_t mech_name,
+        gss_buffer_t mech_description) nogil
+
+    OM_uint32 gss_inquire_mech_for_saslname(
+        OM_uint32 *min_stat,
+        const gss_buffer_t sasl_mech_name,
+        gss_OID *mech_type) nogil
+
+
+def inquire_saslname_for_mech(OID mech):
```

Comment:
RFC 5801 doesn't mark `mech` as optional, so this should be marked `not None` -- it shouldn't be an option to pass no mech.